be/src/Project/Web/Buyer/Buyer.Web/Shared/ModelBuilders/MainNavigations/MainNavigationModelBuilder.cs

```diff
@@ -4,87 +4,86 @@
 using Foundation.PageContent.Components.Links.ViewModels;
 using Foundation.PageContent.ComponentModels;
 using System.Threading.Tasks;
-using Microsoft.Extensions.Localization;
-using Foundation.Localization;
-using Microsoft.AspNetCore.Routing;
-using System.Globalization;
 using Microsoft.Extensions.Options;
 using Buyer.Web.Shared.Configurations;
+using Buyer.Web.Shared.Repositories.MainNavigationLinks;
+using Microsoft.AspNetCore.Routing;
+using System.Globalization;
+using Foundation.Localization;
+using Microsoft.Extensions.Localization;
+using Foundation.Extensions.ExtensionMethods;
 
 namespace Buyer.Web.Shared.ModelBuilders.MainNavigations
 {
     public class MainNavigationModelBuilder : IAsyncComponentModelBuilder<ComponentModelBase, MainNavigationViewModel>
     {
-        private readonly IStringLocalizer<GlobalResources> globalLocalizer;
-        private readonly IStringLocalizer<OrderResources> orderLocalizer;
-        private readonly LinkGenerator linkGenerator;
-        private readonly IOptionsMonitor<AppSettings> settings;
+        private readonly IStringLocalizer<GlobalResources> _globalLocalizer;
+        private readonly IStringLocalizer<OrderResources> _orderLocalizer;
+        private readonly LinkGenerator _linkGenerator;
+        private readonly IOptionsMonitor<AppSettings> _settings;
+        private readonly IMainNavigationLinkRepository _mainNavigationLinkRepository;
 
         public MainNavigationModelBuilder(
             IStringLocalizer<GlobalResources> globalLocalizer,
             IStringLocalizer<OrderResources> orderLocalizer,
+            IOptionsMonitor<AppSettings> settings,
             LinkGenerator linkGenerator,
-            IOptionsMonitor<AppSettings> settings)
+            IMainNavigationLinkRepository mainNavigationLinkRepository)
         {
-            this.globalLocalizer = globalLocalizer;
-            this.orderLocalizer = orderLocalizer;
-            this.linkGenerator = linkGenerator;
-            this.settings = settings;
+            _globalLocalizer = globalLocalizer;
+            _orderLocalizer = orderLocalizer;
+            _settings = settings;
+            _linkGenerator = linkGenerator;
+            _mainNavigationLinkRepository = mainNavigationLinkRepository;
         }
 
         public async Task<MainNavigationViewModel> BuildModelAsync(ComponentModelBase componentModel)
         {
+            var mainNavigationLinks = await _mainNavigationLinkRepository.GetMainNavigationLinksAsync(componentModel.ContentPageKey, componentModel.Language, _settings.CurrentValue.DefaultCulture);
```

Comment:
**suggestion:** let's move it after links, close to where it's used (just before foreach)